controller/node_controller.go

```diff
@@ -1954,3 +1859,117 @@ func isNodeOrDisksEvictionRequested(node *longhorn.Node) bool {
 
 	return false
 }
+
+func (nc *NodeController) setReadyAndSchedulableConditions(node *longhorn.Node, kubeNode *corev1.Node,
+	managerPods []*corev1.Pod) error {
+	nodeReady := true
+	defer func() {
+		// Only record true if we find no reason to record false.
+		if nodeReady {
+			node.Status.Conditions = types.SetConditionAndRecord(node.Status.Conditions,
+				longhorn.NodeConditionTypeReady, longhorn.ConditionStatusTrue,
+				"", fmt.Sprintf("Node %v is ready", node.Name),
+				nc.eventRecorder, node, corev1.EventTypeNormal)
+		}
+	}()
+
+	nodeManagerFound := false
+	for _, pod := range managerPods {
+		if pod.Spec.NodeName == node.Name {
+			nodeManagerFound = true
+			podConditions := pod.Status.Conditions
+			for _, podCondition := range podConditions {
+				if podCondition.Type == corev1.PodReady {
+					if podCondition.Status != corev1.ConditionTrue || pod.Status.Phase != corev1.PodRunning {
+						nodeReady = false
+						node.Status.Conditions = types.SetConditionAndRecord(node.Status.Conditions,
+							longhorn.NodeConditionTypeReady, longhorn.ConditionStatusFalse,
+							string(longhorn.NodeConditionReasonManagerPodDown),
+							fmt.Sprintf("Node %v is down: the manager pod %v is not running", node.Name, pod.Name),
+							nc.eventRecorder, node, corev1.EventTypeWarning)
+					}
+					break
+				}
+			}
+			break
+		}
+	}
```

Comment:
nit: abstract the codes to `checkNodeManagerPodStatus` function

Reply:
Please check the latest commit to see if it meets your request.